crates/sui-types/src/gas_model/gas_v2.rs

```diff
@@ -307,7 +309,10 @@ mod checked {
             // 3. Gas balance (all gas coins together) is bigger or equal to budget
             let mut gas_balance = 0u128;
             for gas_obj in gas_objs {
-                gas_balance += gas::get_gas_balance(gas_obj)? as u128;
+                // expect is safe because we already checked that all gas objects have an address owner
+                gas_balance +=
+                    gas::get_gas_balance(gas_obj.as_object().expect("object must be owned"))?
```

Comment:
I'm not sure if this is 100% correct. In particular, if `gas_object` is a deleted shared object, then according to the check on line 288 `gas_object.as_object()` will be `None`. However, in that case as far as I can see we don't raise an error, so such an object could progress through to here where the `expect` will fail.

Reply:
no, we already test that as_object() returns an object at the top of the function

Reply:
as discussed in slack, i see what you mean, and fixed it